doc/rbd/rbd-live-migration.rst

```diff
@@ -0,0 +1,150 @@
+======================
+ Image Live-Migration
+======================
+
+.. index:: Ceph Block Device; live-migration
+
+RBD images can be live-migrated between different pools within the same cluster
+or between different image formats and layouts. When started, the source image
+will be deep-copied to the destination image, pulling all snapshot history and
+optionally keeping any link to the source image's parent to help preserve
+sparseness.
+
+This copy process can safely run in the background while the new target image is
+in-use. There is currently a requirement to temporarily stop using the source
+image before preparing a migration. This helps to ensure that the client using
+the image is updated to point to the new target image.
+
+.. note::
+   Image live-migration requires the Ceph Nautilus release or later. The krbd
+   kernel module does not support live-migration at this time.
+
+
+.. ditaa:: +-------------+               +-------------+
+           | {s} c999    |               | {s}         |
+           |  Live       | Target refers |  Live       |
+           |  migration  |<-------------*|  migration  |
+           |  source     |  to Source    |  target     |
+           |             |               |             |
+           | (read only) |               | (writable)  |
+           +-------------+               +-------------+
+
+               Source                        Target
+
+The live-migration process is comprised of three steps:
+
+#. **Prepare Migration:** The initial step creates the new target image and
+   cross-links the source and target images. Similar to `layered images`_,
+   attempts to read uninitialized extents within the target image will
+   internally redirect the read to the source image, and writes to
+   uninitialized extents within the target will internally deep-copy the
+   overlapping source image block to the target image.
+
+
+#. **Execute Migration:** This is a background operation that deep-copies all
+   initialized blocks from the source image to the target. This step can be
+   run while clients are actively using the new target image.
+
+
+#. **Finish Migration:** Once the background migration process has completed,
+   the migration can be committed or aborted. Committing the migration will
+   remove the cross-links between the source and target images, and will
+   remove the source image. Aborting the migration will remove the cross-links,
+   and will remove the target image.
+
+Prepare Migration
+=================
+
+The live-migration process is initiated by running the `rbd migration prepare`
+command, providing the source and target images::
+
+    $ rbd migration prepare migration_source migration_target
+
```

Comment:
May be worth noting that `migration_target` (name) may be the same as `migration_source`, which is useful when one just want to convert image (e.g. to different format or layout). And in this case the `migration_target` can be skipped. Without this it might be not clear for users.

Reply:
Done